### PR TITLE
fix: Changing display name [WEB-740]

### PR DESCRIPTION
**Pull request by gt2345**
## Description

After changing the user display name to a different display name, it is impossible for a user to rename their new display name to a previous display name (if previous display name is the same as username).

## Test Plan

Navigate to settings page
Change display name
Change display name back
Test with admin/non-admin user
When test with admin user, set display name to 'admin', change to something else, then change back

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by keita-determined — APPROVED** [review]
could you also remove regex? https://github.com/determined-ai/determined/blob/3fcb9853194113246863f711cb06f25b508548c3/webui/react/src/pages/Settings/SettingsAccount.tsx#L61
i dont think its necessary. 

---

If it's easy to change, could you change the more meannigful error message? https://github.com/determined-ai/determined/blob/9f76ecbc3f44251e3455b18da53173e4edb62701/webui/react/src/pages/Settings/SettingsAccount.tsx#L15 
Error message doesnt tell us why its error. server error? permission error? etc
If its not easy, no need to

---

and need to change commit message or add a new commit to pass ci.

**Comment by gt2345** [comment]
> could you also remove regex?
> 
> https://github.com/determined-ai/determined/blob/3fcb9853194113246863f711cb06f25b508548c3/webui/react/src/pages/Settings/SettingsAccount.tsx#L61
> 
> 
> i dont think its necessary.
> If it's easy to change, could you change the more meannigful error message?
> 
> https://github.com/determined-ai/determined/blob/9f76ecbc3f44251e3455b18da53173e4edb62701/webui/react/src/pages/Settings/SettingsAccount.tsx#L15
> 
> 
> Error message doesnt tell us why its error. server error? permission error? etc
> If its not easy, no need to
> and need to change commit message or add a new commit to pass ci.

@keita-determined 
Dynamic error message added. 
Maybe we only want to allow letters and numbers in display name?

File: webui/react/src/pages/Settings/SettingsAccount.tsx [hunk 1]
```diff
@@ -45,7 +44,7 @@ const SettingsAccount: React.FC = () => {
         updateUser(user.id, (oldUser) => ({ ...oldUser, displayName: newValue }));
         message.success(API_DISPLAYNAME_SUCCESS_MESSAGE);
       } catch (e) {
-        message.error(API_DISPLAYNAME_ERROR_MESSAGE);
+        message.error((e as DetError).message);
         handleError(e, { silent: true, type: ErrorType.Input });
```
Comment (hamidzr) [inline webui/react/src/pages/Settings/SettingsAccount.tsx:48]: nit: I think this should pop up an error message if it's not silenced

**Review by hamidzr — APPROVED** [review]
lgtm. if you want to add an e2e pytest it's decently easy now let me know and I can help or you could probably do a integ test in go as well (might need a bit more set up)

File: webui/react/src/pages/Settings/SettingsAccount.tsx (hunk 1 above)
Comment (gt2345) [reply webui/react/src/pages/Settings/SettingsAccount.tsx:48]: Right! Thanks!

**Pull request merged**